dependencies: Clean up settings_config/page_params/i18n/show_email/etc.

showell opened the pull request.

timabbott commented:
@showell oops, you'll need to rebase this since it conflicts with the `_.extend` removal PR that I just merged.  Should be straightforward if annoying (a large fraction of lines conflict).

timabbott commented:
I'm a little worried about this path:
* It seems like the getters change results in us constructing a new.  What's the issue with just having i18n initialized?  It seems like adding a few "init_i18n" lines in unit tests isn't a big deal.  Or, as @andersk suggests, we could maybe convert `i18n` to be a module and then just require it?
* What's the benefit in making `presence` not access `page_params` directly, instead being passed it? I think it obscures the code to have multiple names for a global.  What do you think about doing conceptually `presence.initialize(page_params.pop('presences', params.pop(''initial_servertime"))` as the pattern for changing initialization methods that we don't want to use `page_params` anymore?
* We should probably be extracting new leaf modules as ES6 modules using the `export` keyword, so we don't have to migrate them later.

showell commented:
I'm gonna abandon this and split it up into new PRs.  This PR had two fairly separate goals:

* Split out `show_email`.  I'll address the objections raised here in that PR.
* Clean up `page_params` initialization.  I agree 100% with Tim's pushback here and think I have a pretty nice way to express what he's going for.

(The only thing that tied the two efforts together was that they, in tandem, eventually purged `page_params` from the `people` code, but that's actually a fairly minor goal and shouldn't actually cause sequencing issues for the merge, I believe.)
